nexus/src/context.rs

```diff
@@ -158,14 +158,27 @@ impl ServerContext {
             nexus_config::Database::FromUrl { url } => url.clone(),
             nexus_config::Database::FromDns => {
                 info!(log, "Accessing DB url from DNS");
-                let address = resolver
-                    .lookup_ipv6(ServiceName::Cockroach)
-                    .await
-                    .map_err(|e| format!("Failed to lookup IP: {}", e))?;
+                let address = loop {
+                    match resolver
+                        .lookup_socket_v6(ServiceName::Cockroach)
+                        .await
+                    {
+                        Ok(address) => break address,
+                        Err(e) => {
+                            warn!(
+                                log,
+                                "Failed to lookup cockroach address: {e}"
+                            );
+                            tokio::time::sleep(std::time::Duration::from_secs(
+                                1,
+                            ))
+                            .await;
+                        }
+                    }
+                };
```

Comment:
I'd kinda like to land somewhere that:

- Whenever we want to access a service, we *could* look up the address again through DNS
- Whenever we want to access DNS, we *could* access any of the DNS server
- If we time-out or error accessing anything, we can customize a back-off policy

... which sorta seems like a Rust version of cueball. Maybe that's a natural follow-up from here?

Reply:
Yes, I think so.  Related to that: we want to proactively resolve stuff _before_ we need it so that DNS resolution latency/availability does not affect latency/availability of the stuff that needs it so much.

Maybe file an issue and add a reference both here and in the Dendrite loop (so that people don't, like, keep copying and pasting this pattern)?